### PR TITLE
Add note on how public CI is impl and what users can do to help.

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

File: docs/source/developer/contributing.rst [hunk 1]
```diff
@@ -279,22 +279,71 @@ This translates into the fact that the test suite passes without errors
...
+hardware!
+
+Helping with Public CI load
+'''''''''''''''''''''''''''
+
+Even with the mitigating design above public CI can get overloaded which causes
+a backlog of builds. It's therefore really helpful when opening pull requests if
+you can limit the frequency of pushing changes. Ideally, please squash commits
+to reduce the number of patches and/or push as infrequently as possible. The
+core developers thank everyone for their cooperation with this!
```
Comment (seibert) [inline docs/source/developer/contributing.rst:346]: Should we remind people not to rebase or force push squashed commits here as it messes up the code review history?

Comment (stuartarchibald) [reply docs/source/developer/contributing.rst:346]: Excellent point, yes, will add!

Comment (stuartarchibald) [reply docs/source/developer/contributing.rst]: Done in 482bdb9

**Review by esc — APPROVED** [review]
LGTM! I'll leave it in the Ready for Review state for now in case anyone else would like to review this.

**Comment by esc** [comment]
@sklam do you wan to take a look at this too? If not, I think we can mark as RTM!

File: docs/source/developer/contributing.rst [hunk 2]
```diff
@@ -279,22 +279,75 @@ This translates into the fact that the test suite passes without errors
...
 AMD and NVIDIA GPUs.  The build system however is internal to Anaconda, so we
-also use `Travis CI <https://travis-ci.org/numba/numba>`_ and
-`Azure <https://dev.azure.com/numba/numba/_build>`_ to provide public continuous
-integration information for as many combinations as can be supported by the
-service.  Travis CI automatically tests all pull requests on OS X and Linux, as
-well as a sampling of different Python and NumPy versions, Azure does the same
-but also includes Windows.  If you see problems on platforms you are unfamiliar
-with, feel free to ask for help in your pull request.  The Numba core developers
-can help diagnose cross-platform compatibility issues.
+also `Azure <https://dev.azure.com/numba/numba/_build>`_ to provide public
```
Comment (sklam) [inline docs/source/developer/contributing.rst:290]:
```suggestion
also use `Azure <https://dev.azure.com/numba/numba/_build>`_ to provide public
```

**Review by sklam — COMMENTED** [review]
just one minor thing

**Comment by esc** [comment]
@sklam @stuartarchibald thanks for fixing this up, marking as RTM now.

**Comment by stuartarchibald** [comment]
Thanks for the review.

**Pull request merged**